### PR TITLE
Fix linting and tests

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Changes:
- add lower and upper bounds to several development tools and depdendencies in Pipfile and tox.ini: black, isort, flake8, pytest, etc.
- (Temporarily) remove darglint checker from the linting dev workflow
- Update Pipfile.lock according to the previous point
- Run black against the source code

## Fixes

n/a

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/master/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

## Further comments

n/a

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -96,6 +96,10 @@ def markdown_parser():
 
 #################################################
 # Import PDDL fixtures
+from tests.fixtures.code_objects.blocksworld_fond import (  # noqa: E402, F401
+    blocksworld_fond_01,
+    blocksworld_fond_domain,
```
Comment (haz) [inline tests/conftest.py:101]: Order of these meant to change?

**Review by haz — CHANGES_REQUESTED** [review]
Only one little thing I noticed.

File: tests/conftest.py (hunk 1 above)
Comment (haz) [reply tests/conftest.py:101]: Nevermind...this is just an import and not a function call.

**Review by haz — APPROVED** [review]

File: tests/conftest.py (hunk 1 above)
Comment (marcofavorito) [reply tests/conftest.py:101]: `isort` is the guilty here

**Pull request merged**